### ISSUE TITLE
refactor: type system

### ISSUE THREAD

**Issue by ltnscp9028** [Commit-Wizard/commit-wizard#17]
In this file, handleBranchComparison parameter input type is string.
https://github.com/Commit-Wizard/commit-wizard/blob/main/src/clients/diff-options.generator.ts#L32-L37

But Actuall,y input type must be tuple ```inputs:[string, string]```

AS-IS
```ts
  static handleBranchComparison(inputs: string[]): DiffOptions {
    if (inputs.length !== 2) {
      throw new Error('Branch comparison requires exactly two branch names.');
    }
    return { type: 'branch', branch1: inputs[0], branch2: inputs[1] };
  }
```

TO-BE
```ts
  static handleBranchComparison(inputs:[string, string]): DiffOptions {
    if (inputs.length !== 2) {
      throw new Error('Branch comparison requires exactly two branch names.');
    }
    return { type: 'branch', branch1: inputs[0], branch2: inputs[1] };
  }
```

**Comment by gimseonjin** [comment]
Hi there,

Thank you for bringing this to our attention. You're absolutely right, the input type for handleBranchComparison should indeed be a tuple [string, string] to accurately reflect the requirement for exactly two branch names. This will ensure that the function's intent is clear and that TypeScript can enforce the correct usage at compile time, providing a more robust type check.

I have reviewed the current implementation and agree with your suggestion. We'll proceed with updating the method signature to inputs: [string, string] as proposed. This change will help prevent incorrect calls to handleBranchComparison and clarify the expected input.

We appreciate your contribution to improving the codebase, and I'll make the necessary changes shortly. Please feel free to review the update once it's committed and continue to share any further feedback or suggestions you may have.

Thanks again for your support and keen eye for detail!

**Comment by gimseonjin** [comment]
We've addressed this issue in version 0.4.5! Thank you for bringing it to our attention and for your patience. Please update to the latest version to benefit from this fix.
